### DISCUSSION TITLE
Change the creator of a page

### DISCUSSION THREAD

**Discussion by mittkkoo** [requarks/wiki#6393] [Help / Questions] [answered]
I have the funny situation where some people don't want to write in the wiki, but want to be listed as the page creator for specific pages. 

The question that arises is how to change the creator of a page after it has been created? I'm open to workarounds as well.

**Comment by fnafox** [comment]
Tell them they can't have laurels for someone else's work.

More seriously, I don't think there's a way to do it other than directly altering the field in the database.
Check the table pageHistory, I'd imagine authorId is the id of the user in question.

**Comment by NGPixel** [comment, marked as answer]
Editing the database directly is your only option.

I don't see adding this option in the UI any time soon as this is a very niche scenario and doesn't make much sense IMO.

**Comment by mittkkoo** [comment]
Thank you for the hints!

With a PostgreSQL database, here is how I managed to modify the "Created by" entry in the page history for page with id=44 to show the user with user id=14.

```c
wiki=# UPDATE "public"."pageHistory" 
SET "authorId" = 14 
WHERE "pageId" = 44 
AND "createdAt" = (select min("createdAt") FROM "public"."pageHistory" WHERE "pageId" = 44);
UPDATE 1
```

And it seems that the real creator of the page (for the example here this was user id=1) is still saved in another entry here, but this doesn`t seem to be visible anywhere in the wiki:
```c
wiki=# SELECT "creatorId" FROM pages WHERE id= 44; 
-[ RECORD 1 ] 
creatorId | 1 
```
